### PR TITLE
feat: modal which shows in which form the question is used

### PR DESCRIPTION
**Pull request by StephanH90**
## Description

This MR introduces a new modal which shows in which form a question is being used (and if that form is archived or not published).

#### Button which opens the modal
image

(I'm not sure about the label)

#### Modal
image

### CODE REVIEW

**Comment by StephanH90** [comment]
@anehx y u no review?

File: packages/form-builder/addon/components/cfb-form-editor/question-usage.hbs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  <modal.header>
+    {{t "caluma.form-builder.question.usage.title"}}
+  </modal.header>
+  <modal.body>
+    {{#if this.forms.isLoading}}
+      <div class="uk-text-center">
+        <UkSpinner @ratio={{1}} />
+      </div>
+    {{else}}
+      <ul class="uk-list uk-list-divider" id="question-list">
```
Comment (anehx) [inline packages/form-builder/addon/components/cfb-form-editor/question-usage.hbs:24]: What's the ID for?

**Review by anehx — CHANGES_REQUESTED** [review]

**Comment by czosel** [comment]
Minor UX idea: Instead of the button, we could display a link _only if the question is used in more than one form_:

> This question is used in (n) other forms.

File: packages/form-builder/translations/de.yaml [hunk 2]
```diff
@@ -133,6 +133,12 @@ caluma:
 
       hideLabel: "Label verstecken"
 
+      usage:
+        button: "Formulare mit dieser Frage"
+        title: "Formulare mit dieser Frage"
```
[the remark this answers is not in the document]
Comment (derrabauke) [reply packages/form-builder/translations/de.yaml:138]: I've merge the tow into one `title` translation.

File: packages/form-builder/translations/de.yaml [hunk 3]
```diff
@@ -133,6 +133,12 @@ caluma:
...
+        no-forms-found: "Es wurde keine Formular gefunden welche diese Frage verwenden"
```
[the remark this answers is not in the document]
Comment (derrabauke) [reply packages/form-builder/translations/de.yaml:139]: `done`

**Comment by derrabauke** [comment]
> Minor UX idea: Instead of the button, we could display a link _only if the question is used in more than one form_:
> 
> > This question is used in (n) other forms.
> 
> Or, if you prefer the button:
> 
> > Forms with this question (n)
> 
> where n >= 2

I like the idea. That's how it looks now and I think the link is less obtrusive than the button.
image

File: packages/form-builder/addon/components/cfb-form-editor/question-usage.hbs [hunk 4]
```diff
@@ -0,0 +1,56 @@
+<UkButton
+  {{on "click" (fn (mut this.modalVisible) true)}}
+  data-test-show-question-usage-modal-btn
+>
+  {{t "caluma.form-builder.question.usage.button"}}
+</UkButton>
```
[the remark this answers is not in the document]
Comment (derrabauke) [reply packages/form-builder/addon/components/cfb-form-editor/question-usage.hbs:6]: See https://github.com/projectcaluma/ember-caluma/pull/2467#issuecomment-1860951813

File: packages/form-builder/addon/components/cfb-form-editor/question-usage.hbs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  data-test-question-usage-modal
+  as |modal|
+>
+  <modal.header>
+    {{t "caluma.form-builder.question.usage.title"}}
+  </modal.header>
+  <modal.body>
+    {{#if this.forms.isLoading}}
+      <div class="uk-text-center">
+        <UkSpinner @ratio={{1}} />
```
[the remark this answers is not in the document]
Comment (derrabauke) [reply packages/form-builder/addon/components/cfb-form-editor/question-usage.hbs:21]: `removed`

File: packages/form-builder/addon/components/cfb-form-editor/question-usage.hbs (hunk 1 above)
Comment (derrabauke) [reply packages/form-builder/addon/components/cfb-form-editor/question-usage.hbs:24]: I think it was a copy-paste-relict. Couldn't find any usage for it.

**Comment by derrabauke** [comment]
* *active* state UI:
image

* *inactive* (no refs) state UI:
image

**Comment by czosel** [comment]
Im not sure if the inactive (no refs) state is worth it. The label would have to be „this question is not used in any _other_ form“, since it is always used in at least one form (the one that’s currently open). And that label sounds a bit complicated and doesn’t add much value, does it?

**Comment by derrabauke** [comment]
> Im not sure if the inactive (no refs) state is worth it. [...] and doesn’t add much value, does it?

I'm also unsure, but I wanted to cover all the possible cases. :shrug:

**Comment by derrabauke** [comment]
> The label would have to be „this question is not used in any _other_ form“, since it is always used in at least one form (the one that’s currently open). And that label sounds a bit complicated and doesn’t add much value, does it?

To condense it to the main idea of the feature: We want to show a hint in case this question is used in any _other_ form. And if so we want to show all the references in a modal. Right?

So the **link** should sound: "The question is used in _n_ *other* form(s)." -> excluding the current form, since that's a given.
And the **modal**: "All references of this question" -> including the current form.

**Comment by derrabauke** [comment]
Updated design:
* link
image

* modal 
image

File: packages/form-builder/addon/components/cfb-form-editor/question-usage.hbs [hunk 6]
```diff
@@ -0,0 +1,52 @@
+{{#let this.forms.value as |forms|}}
```
Comment (anehx) [inline packages/form-builder/addon/components/cfb-form-editor/question-usage.hbs:1]: Not really necessary as you only need it once on L25

**Review by anehx — CHANGES_REQUESTED** [review]

**Pull request merged**